holistics/holistics.py

```diff
@@ -89,11 +82,25 @@ def DownloadResults(self):
                 try:
                     data.to_csv(self.path, encoding='utf-8', index=False)
                     print ("Export to file successed!")
+                    return 1
                 except:
                     print ("Can't export file to " + self.path)
+        except requests.exceptions.HTTPError as err:
+            return err
         
 
-    def ExportData(self,report_id,filters=None, page_size=None, page=None):
-        self.SubmitReport(report_id, filters)
-        self.GetExportResults(self.page['job_id'],page_size,page)
-        return self.DownloadResults()
+    def export_data(self,report_id,filters=None, page_size=None, page=None):
+        res = self.submit_report(report_id, filters)
+        if type(res) == requests.exceptions.HTTPError:
```

Comment:
not a proper way to handle Exceptions. Exceptions must be handled by try..except..  
If there is an error, the function having the error must raise the Exception, not returning one (see comment for `submit_report` function).  
Then, in this function we can do something like
```python
try:
    res = self.submit_report(report_id, filters)
    res = self.get_export_result(res,page_size,page)
    res = self.download_results()
    return res
except HolisticsException as e:
    # print error message and return 0 just to follow current behaviour. In fact we do not need to catch this exception, we can just let the caller of `export_data` handle it
    print (e.message)
    return 0
```

Reply:
If you dont want to wrap error in `HolisticsException`, remove exception catching in all inner functions and do something like:
```python
try:
    res = self.submit_report(report_id, filters)
except requests.exceptions.HTTPError as e:
    # handle error
try:
    res = self.get_export_result(res,page_size,page)
except requests.exceptions.HTTPError as e:
    # handle error
...
```